crates/autopilot/src/run_loop.rs

```diff
@@ -553,6 +553,24 @@ impl RunLoop {
             std::cmp::Reverse(participant.solution().score().get().0)
         });
 
+        // Filter out solutions that don't come from their corresponding submission
+        // address
+        let solutions = solutions
+            .into_iter()
+            .filter(|participant| {
+                let submission_address = participant.driver().submission_address;
+                let is_solution_from_driver = participant.solution().solver() == submission_address;
+                if !is_solution_from_driver {
+                    tracing::warn!(
+                        driver = participant.driver().name,
+                        ?submission_address,
+                        "the solution is not received from the driver submission address"
```

Comment:
I assume, the solution is still received.
```suggestion
                        "the solution received is not from the driver submission address."
```